### DISCUSSION TITLE
Reload/Unload/Start/Restart config parsing.

### DISCUSSION THREAD

**Discussion by kopavel** [davmac314/dinit#155] [Ideas]
Instead of unloading service isn't more preferable just re-parse service config on every start?
When write and debug new service it may be confusing changing config and not see any difference (if forgot unloading service manually, not just restart it).

**Comment by davmac314** [comment]
This was a deliberate decision. First, one general principle that dinit mostly adheres to is "don't do things implicitly". When every action and setting is explicit, there is less potential surprise and the user's familiarity with the system is increased.

While I understand that it might be confusing that your changes aren't applied unless you tell dinit to reload the service, you could make the same argument that changes should be applied even to a running service if the configuration is edited. And that is clearly going too far, by my judgement. 

Not automatically reloading services has advantages:
- the service description set as a whole can be modified without worrying that a service will restart in the middle of the operation and use the new configuration that isn't complete or that is intended only for the next reboot
-  faster, especially if it means a disk doesn't have to be spun up

I would certainly accept a PR for a feature where dinitctl warns if the service description has been changed for example.

**Reply by kopavel** [reply]
The last one opinion are acceptable - that's how react systemd for that situation - warn about config change, and suggest to reload with  provided command example. 
I agree that it's my bad - I not read whole man carefully about dinitctl and only find about reload behavior in a middle of my experiments after another rereading of documentation.
